images/linux/scripts/installers/python.sh

```diff
@@ -10,47 +10,9 @@ source $HELPER_SCRIPTS/document.sh
 # Install Python, Python 3, pip, pip3
 apt-get install -y --no-install-recommends python python-dev python-pip python3 python3-dev python3-pip
 
-# Install PyPy 2.7 to $AGENT_TOOLSDIRECTORY
-wget -q -P /tmp https://bitbucket.org/pypy/pypy/downloads/pypy2.7-v7.1.0-linux64.tar.bz2
-tar -x -C /tmp -f /tmp/pypy2.7-v7.1.0-linux64.tar.bz2
-rm /tmp/pypy2.7-v7.1.0-linux64.tar.bz2
-mkdir -p $AGENT_TOOLSDIRECTORY/PyPy/2.7.13
-mv /tmp/pypy2.7-v7.1.0-linux64 $AGENT_TOOLSDIRECTORY/PyPy/2.7.13/x64
-touch $AGENT_TOOLSDIRECTORY/PyPy/2.7.13/x64.complete
-
-# add pypy to PATH by default
```

Comment:
We'll want to keep this and the one for PyPy3 below.  Unfortunately that adds something to keep in sync when we update PyPy -- not sure if there's a better way.  Python will have the same problem.

Reply:
I think a good place for it would be `hosted-tool-cache.sh` where you are documenting it now.

Reply:
In the `setup.sh` file that gets downloaded and run, there are symlinks that get created.  
```
# PEP 394-like symlinks (but don't include minor version)
ln -s pypy python2
ln -s python2 python
```
Couldn't we add an extra symlink to `setup.sh`? We need this so that `pypy` can be used without the `setup-python` action?
```
ln -s pypy /usr/local/bin/pypy
```

Reply:
As long as there are only 2 versions of PyPy in the tools cache then there shouldn't be any conflicts. For Python this will have to be addressed differently

Reply:
> The idea of having to make a change here every time there is a version update in the tools cache just doesn't seem right

That's a fair point.  See [how we do it for Windows](https://github.com/actions/virtual-environments/blob/07b9e016112891bbec8108d7dc941a84aeb8989b/images/win/scripts/Installers/Windows2019/Install-Python.ps1#L27) currently.  So whichever pattern you use, I think it should be consistent.

I think that should work -- you might have to `sudo` to write to `/usr/local/bin`.  I don't see what the problem with CPython would be, could you explain?

Reply:
Based off of the options, I decided to add `pypy` and `pypy3` to PATH in `hosted-tool-cache.sh`. This was done largely to be consistent with how Windows does it. Also contemplated doing something like  `export PATH`, but decided against that. In the future we will be able to completely remove `python.sh` 🔥 when the extra versions of python that get downloaded get removed.

The change I made runs on the assumption that there will every only be 2 versions of PyPy in the tools cache (which i think is safe to have). 

Regarding CPython I was originally thinking of modifying the `setup.sh` file that gets downloaded and having the link get created there as one of the steps. Modifying `setup.sh` and adding the symlink there would work for PyPy because there would only be 2 versions downloaded. With Python there are a lot more than 2 so this would cause a problem with what becomes the default?. For CPython, if the symlink gets created just like PyPy now (some if statement), then we can choose which version of Python is the default one that gets added to PATH.

Reply:
> The change I made runs on the assumption that there will every only be 2 versions of PyPy in the tools cache

There aren't any plans right now to have more versions, and if there are we can always change this.  Thanks for calling out the assumption, though.